lib/Serialization/Deserialization.cpp

```diff
@@ -3723,12 +3727,14 @@ class swift::DeclDeserializer {
     while (auto *genericParams = MF.maybeReadGenericParams(DC)) {
       genericParams->setOuterParameters(outerParams);
 
-      // We do this repeatedly to set up the correct DeclContexts for the
-      // GenericTypeParamDecls in the list.
-      extension->setGenericParams(genericParams);
+      // Set up the DeclContexts for the GenericTypeParamDecls in the list.
+      for (auto param : *genericParams)
+        param->setDeclContext(extension);
```

Comment:
This is part of your GenericParamListRequest::cacheResult() so it's redundant here

Reply:
It's not redundant for the inner parameter lists - the old code was pumping `setGenericParams` just for this effect.

Reply:
Perhaps later we can investigate deserializing the generic parameter list after we deserialize its owner decl, giving generic parameter decls the right DeclContext upfront. At one point there were circularity issues though